### PR TITLE
Feat/1692 data file upload bug

### LINKED ISSUE #1692: As a non-ACF OCIO user, I need the ability to submit data files
Thank you for taking the time to let us know about the issue you found. The basic rule for bug reporting is that 
something isn't working the way one would expect it to work. Please provide us with the information requested 
below and we will look at it as soon as we are able.

## Description

I no longer have the ability to submit data files as a `Data Analyst`, `OFA Admin`, `OFA Regional Staff`, and `Developer` . I noticed this during testing for #1658, since I need to submit a data file to confirm on the backend that the metadata for the file is read-only. 

## Action Taken

1. sign in @ https://tdp-frontend-staging.app.cloud.gov/
2. confirm on backend that you are assigned to a permissions group that should be able to submit data files. 
3. navigate to frontend and look for `Data Files` tab. 

## What I expected to see

bug7
 (interestingly, this is visible for `OFA System Admin` but not usable because this role cannot have an STT set due to #1387.)

## What I did see

_data analyst role_
bug1
bug2
bug3

_OFA admin role_
bug4
bug5
bug6

## Other Helpful Information

+ URL of the page I was on: tested  `staging` and `raft` deployed sites
+ Browser and version: Version 96.0.4664.93 (Official Build) (32-bit)
+ Operating System: Windows 
+ Is the issue repeatable?: yes
+ Has the issue occurred more than once?: yes

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes

Pull request closes #1692 

## How to Test

```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Make sure your account is one of the ones expected to have Data file access
3. is there a Data files tab?

image

4. do so for every role expected to have data files access (Which are these:
'Developer', 'Data Analyst', 'Regional Staff', 'OFA Admin')

## Tabletop

- [x] Ready for Tabletop (Reviewer check this)
- [x] Tabletop scheduled (Assignee check this)
- [ ] Tabletop approval 

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x]  Developers can see data files nav button
+ [x]  Data Analysts can see data files nav button
+ [x]  Regional Staff can see data files nav button
+ [x]  OFA Admin can see data files nav button
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?

  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented?

### CODE REVIEW

File: tdrs-frontend/src/components/Header/Header.test.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('Header', () => {
     auth: {
       user: {
         email: 'test@test.com',
-        roles: [{ id: 1, name: 'OFA System Admin', permissions: [] }],
+        roles: [{ id: 1, name: 'Developer', permissions: [] }],
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/Header/Header.test.js:17]: this is the only role that's suppose to have both permission to data file and permission to admin

Comment (riatzukiza) [reply tdrs-frontend/src/components/Header/Header.test.js:17]: open to more tests, though we are still covered over 90% according to code cov

**Comment by abottoms-coder** [comment]
While the `developer` role can see the data files tab and page, they cannot use it because I don't believe they can be assigned a location within the Django Admin Console. This breaks the AC of #1692

**Review by abottoms-coder — CHANGES_REQUESTED** [review]
While this addresses frontend changes, the AC of #1692 suggest we need to alter the Django form validation.

**Review by reitermb — APPROVED** [review]
This is good to go from an a11y perspective!

Testing notes:
- There do seem to still be django user/group validation problems (e.g. it wasn't possible for me to test as a developer or a data analyst because it threw an error whenever a location was submitted) 
- Consequently I tested as an OFA Admin. Uploading/submitting now seems to work without a 500 error! 🚀 
- I did get a 400 error on my first attempt but the actual submission turned out to have worked when I refreshed and I was unable to produce the same error on subsequent attempts. 
- You may notice "no file selected" readouts from Voiceover on/around file inputs—this _is_ an a11y issue but is also outside the scope of this issue. #1107 will deliver a fix for that.

**Comment by abottoms-coder** [comment]
Testing with latest I've found the roles Data Analyst, Developer, OFA Admin and Regional Staff now can see the data files tab and have their stt/region assigned (excluding OFA Admin, rightly). Outside the scope of this ticket, a Regional Staff user can't *use* the data-files tab because they don't have an STT but perhaps this will be resolved in the #1605 epic.

**Review by abottoms-coder — APPROVED** [review]

**Review by jorgegonzalez — APPROVED** [review]

**Comment by ADPennington** [comment]
@riatzukiza test results below. Please double-check `ofa sys admin` and `ofa regional staff`. i think we need a follow-on ticket to address the ideal behavior for the OFA groups, but for now, what shows in the nav bar isnt aligned with what the permission rules are for these groups:

- `OFA System Admin` cannot see data files tab, but should according to the permission rules  (i wouldn't expect them to be able to actually submit a data file given 1387) ❌ 
1699p3

- `OFA Admin` can see data files tab, as expected. and can submit data for any STT ✔️ 
1699p2
- `Developer` can see data files tab, as expected. and can submit data for STT in profile ✔️ 
1699p5
- `Data analyst` can see data files tab, as expected. and can submit data for STT in profile ✔️ 
1699p4

- `OFA Regional Staff` cannot see data files tab, but should according to the permission rules (same note as above re: sys admin). ❌ 
1699p1

- `ACF OCIO` cannot see data files tab, as expected. ✔️ 
1699p6

**Review by ADPennington — CHANGES_REQUESTED** [review]
see test results. @riatzukiza

**Comment by ADPennington** [comment]
@riatzukiza thanks for the fix. `ofa regional staff` and ofa system admin` now both see data files tab (below):

1699p8
1699p7

**Review by ADPennington — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**